packages/myst-cli/src/transforms/embed.ts

```diff
@@ -35,13 +36,26 @@ function mutateEmbedNode(
   const { url, dataUrl, targetFile, sourceFile } = opts ?? {};
   if (targetNode && node['remove-output']) {
     targetNode = filter(targetNode, (n: GenericNode) => {
+      // After reduction, 'output' nodes may be replaced by their children which are then tagged as outputs
       return n.type !== 'output' && n.data?.type !== 'output';
     });
   }
   if (targetNode && node['remove-input']) {
-    targetNode = filter(targetNode, (n: GenericNode) => {
-      return n.type !== 'code' || n.data?.type === 'output';
-    });
+    targetNode = filter(
+      targetNode,
+      (
+        n: GenericNode,
+        index: number | null | undefined,
+        parent: GenericNode | undefined | null,
+      ) => {
+        // If we have a code cell with a NotebookCell parent, assume its an input cell and remove.
+        return !(
+          n.type === 'code' &&
+          parent?.type === 'block' &&
+          parent?.kind === NotebookCell.code
+        );
```

Comment:
The previous logic had a special case for code-nodes that were tagged as outputs. Now, we want to be even stricter --- only code nodes that are children of code-cells _and_ not tagged as outputs.